rllib/algorithms/algorithm_config.py

```diff
@@ -2173,6 +2193,75 @@ def debugging(
 
         return self
 
+    def fault_tolerance(
+        self,
+        ignore_worker_failures: Optional[bool] = NotProvided,
+        recreate_failed_workers: Optional[bool] = NotProvided,
+        max_num_worker_restarts: Optional[int] = NotProvided,
+        delay_between_worker_restarts_s: Optional[float] = NotProvided,
+        restart_failed_sub_environments: Optional[bool] = NotProvided,
+        num_consecutive_worker_failures_tolerance: Optional[int] = NotProvided,
+        worker_health_probe_timeout_s: int = NotProvided,
+        worker_restore_timeout_s: int = NotProvided,
+    ):
+        """Sets the config's fault tolerance settings.
+
+        Args:
+            ignore_worker_failures: Whether to attempt to continue training if a worker
```

Comment:
Clarification: Does this will cause the failed workers to not recover? So num_healthy workers will go down upon failure and won't come back up? Can you add a bit of elaboration from this aspect? I think I don't still understand how this will work in combo with recreate_failed_workers. One seems redundant.

Reply:
so my understanding is that we have three possible settings:
1. recreate_failed_workers = True (ignore_failed_Workers will be ignored)
2. recreate_failed_Workers = False but ignore_failed_workers = True, upon failure of the workers, num_healthy workers will go down without any recovery, until you run out of them and the job will hang --> why would anyone want this?
3. recreate_failed_worker = False but ignore_failed_worker = False, upon failure of the worker, the algorithm will imediately crash. This is good for testing the env construction --> good for debugging. 

Based on this argument I don't see any real use-case for 2. So I propose merging 1 and 3 to require only one flag (recreate_failed_worker)

Reply:
ok, let's deprecate it.
your understanding is correct. this was introduced way back when there was no fault tolerance.
so by ignoring worker failures, users' jobs at least can continue.
I don't know why would anyone want this.